lib/lcset.gd

```diff
@@ -0,0 +1,51 @@
+##############################################################################
+##
+#W  lcset.gd                  GAP4 package `Utils'               Chris Wensley
+##
+#Y  Copyright (C) 2015-2022, The GAP Group 
+
+#############################################################################
+##
+#C  IsLeftCosetObj( <obj> ) . . . . . . . . . . . . . . coset of the form gU
+#R  IsLeftCosetDefaultRep
+#V  IsLeftCosetFamily 
+#T  IsLeftCosetType 
+#P  IsLeftCoset 
+##
+DeclareCategory( "IsLeftCosetObj", IsDomain and IsExternalOrbit ); 
+DeclareRepresentation( "IsLeftCosetDefaultRep",
+    IsComponentObjectRep and IsAttributeStoringRep and IsLeftCosetObj, 
+    [ "Representative", "ActingDomain" ] ); 
+BindGlobal( "IsLeftCosetFamily", NewFamily( "IsLeftCosetFamily", 
+            IsMultiplicativeElementWithInverse ) ); 
+BindGlobal( "IsLeftCosetType", 
+            NewType( IsLeftCosetFamily, IsLeftCosetDefaultRep ) ); 
+DeclareProperty( "IsLeftCoset", IsLeftCosetObj ); 
+
+#############################################################################
+##
+#A  AssociatedLeftCoset( <rc> ) . . . . . . . . . . . . . . . . gU <--> Ug^-1
+#A  AssociatedRightCoset( <lc> ) . . . . . . . . . . . . . . .  gU <--> Ug^-1
+##
+DeclareAttribute( "AssociatedLeftCoset", IsRightCoset );
+DeclareAttribute( "AssociatedRightCoset", IsLeftCoset );
```

Comment:
How about dropping these and instead just installing methods for `Inverse` (which is also an attribute)?

Reply:
I like this idea, but have one concern.  The basic property in the definition of inverse states that  x * Inverse(x)  is an identity - but that would not be the case here.  Would this not cause problems down the line?

Reply:
I don't see how... I mean yes, in theory, it could cause problems if someone wanted to implement proper "quotient groups" whose elements are cosets of normal subgroups, and then work in such a group.... But there are tons of other things that would go wrong in that setting. I think in such a setting one needs to rethink things anyway.

Beside, this is "just" the utils package, not the GAP kernel; believe me, there are packages doing far worse... 

But yeah, for inclusion in GAP itself, that's a concern among other one would have to deal with in more depth...

Reply:
OK - will use Inverse, since it seems to work well.